File: src/main/java/by/masarnovsky/releasedemon/service/ArtistService.java
package by.masarnovsky.releasedemon.service;

import by.masarnovsky.releasedemon.entity.Artist;

import java.util.List;
import java.util.Optional;
import java.util.Set;

public interface ArtistService {
    Set<Artist> findAll();

    Optional<Artist> findById(Integer id);

    Artist findByName(String name);

    Artist save(Artist artist);

    List<Artist> saveAll(List<Artist> artists);

    List<Artist> findAllByNameIn(List<String> artist);

    List<Artist> findAllWithMbidId();
}

File: src/main/kotlin/by/masarnovsky/releasedemon/service/ArtistService.kt
package by.masarnovsky.releasedemon.service

import by.masarnovsky.releasedemon.entity.Artist
import by.masarnovsky.releasedemon.repository.ArtistRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.util.*
import kotlin.collections.HashSet

@Service
class ArtistService(
    val repository: ArtistRepository,
) {
    fun findAll(): Set<Artist> {
        return HashSet(repository.findAll())
    }

    fun findByName(name: String): Artist? {
        return repository.findByName(name)
    }

    fun save(artist: Artist): Artist {
        return repository.save(artist)
    }

    fun saveAll(artists: List<Artist>): List<Artist> {
        return repository.saveAll(artists)
    }

    fun findAllByNameIn(artists: List<String>): List<Artist> {
        return repository.findAllByNameIn(artists)
    }

    fun findAllWithMbidId(): List<Artist> {
        return repository.findAllByMbidIsNotNull()
    }
}